static/js/entry/zendesk_widget.js

```diff
@@ -116,6 +117,35 @@ const zendeskCallbacks = {
         }, 100);
       };
     }
+  },
+  ticketSubmissionFormLoaded: () => {
+    const iframe = document.querySelector("iframe.zEWidget-ticketSubmissionForm");
+    // field 24690866 is the MicroMasters program selector
+    const fieldNames = ["name", "email", "24690866"];
+    const fieldHidden = [
+      SETTINGS.user && SETTINGS.user.first_name && SETTINGS.user.last_name,
+      SETTINGS.user && SETTINGS.user.email,
+      SETTINGS.program && SETTINGS.program.slug,
+    ];
+    const fields = _.map(
+      fieldNames,
+      (name) => {
+        return iframe.contentDocument.querySelector(
+          `input[name="${name}"], select[name="${name}"]`
+        );
+      }
+    );
+    _.zip(fields, fieldHidden).forEach((values) => {
+      const [ field, isHidden ] = values;
+      if (isHidden) {
+        const label = field.closest('label');
+        label.style.setProperty("display", "none", "important");
+      }
+    });
```

Comment:
the implementation of this is a bit confusing to me. It seems like what we're doing is that we have three different fields ('name', 'email', and '24690866'), which have differing conditions for when they are hidden.

What do you think about breaking this up into a series of functions? Something like:

```js
const fieldSelector = name => (
  `input[name="${name}"], select[name="${name}"]`
);

const fieldElement = R.compose(
  iframe.contentDocument.querySelector, fieldSelector
);

const fieldVisibility = {
  'key': visibility // true | false
  ...
};


const adjustFieldsVisibility = R.map(name => {
  if ( !fieldVisibility(name) ) {
    const label = fieldElement(name).closest('label');
    label.style.setProperty("display", "none", "important");
  };
});

adjustFieldsVisibility([
  'name', 'email', '24690866'
]);
```